server/src/controllers/utils/courseInstance.ts

```diff
@@ -0,0 +1,23 @@
+// SPDX-FileCopyrightText: 2023 The Aalto Grades Developers
+//
+// SPDX-License-Identifier: MIT
+
+import CourseInstance from '../../database/models/courseInstance';
+
+import { ApiError } from '../../types/error';
+import { HttpCode } from '../../types/httpCode';
+
+/**
+ * Finds a course instance by its ID.
+ * @param {number} id - The ID of the course instance.
+ * @returns {Promise<CourseInstance>} - The found course instance model object.
+ * @throws {ApiError} - If the course instance is not found, it throws an error with a message
+ * indicating the missing course instance with the specific ID.
+ */
+export async function findCourseInstanceById(id: number): Promise<CourseInstance> {
+  const courseInstance: CourseInstance | null = await CourseInstance.findByPk(id);
+  if (!courseInstance) {
+    throw new ApiError(`course instance with ID ${id} not found`, HttpCode.NotFound);
```

Comment:
This should also take the error code as a parameter, since we may not necessarily always want to return 404 when a course instance isn't found.